util/feeding.py

```diff
@@ -109,14 +109,15 @@ def to_sparse_tuple(sequence):
 
 def create_dataset(sources,
                    batch_size,
+                   limit=-1,
```

Comment:
If I remember correctly, the limit was once introduced in combination with an ability to start from a certain sample offset or to reverse the data-set. This allowed us to test for maximum batch size (testing the large samples only) without having to run a full epoch.

We should really make it `None`-able (no limit) and have negative numbers take samples from the end of the collections.

Reply:
> f I remember correctly, the limit was once introduced in combination with an ability to start from a certain sample offset or to reverse the data-set.

Show me the code, because as much as I could find:
 - this has been broken for so long
 - this was basically what I implemented



> We should really make it `None`-able (no limit) and have negative numbers take samples from the end of the collections.

I don't get your point.

Reply:
At the moment samples are taken from the beginning of the collections (they are typically sorted shortest to largest). So a limit of e.g. 10 samples takes the first 10 samples (typically the shortest ones) of each set. In my proposal -10 would allow the user to take the 10 last samples (the largest ones) of each set instead.

Reply:
I think he means make it equivalent to Python slicing syntax: `--limit N` means taking `dataset[:N]`. But with negative values this would not be useful for testing maximum batch sizes. For that you'd have to do `dataset[-N:]`.

Reply:
> I think he means make it equivalent to Python slicing syntax: `--limit N` means taking `dataset[:N]`. But with negative values this would not be useful for testing maximum batch sizes. For that you'd have to do `dataset[-N:]`.

`Interleaved` does not support `reversed()` and `itertools.islice` does not support negative values. Since we `yield` from `SDB` or `CSV` classes within `__iter__` I don't see how we can handle negative values to pull from the end of the dataset without being more intrusive into those classes.